include/deal.II/grid/grid_generator.h

```diff
@@ -1907,6 +1907,31 @@ namespace GridGenerator
   flatten_triangulation(const Triangulation<dim, spacedim1> &in_tria,
                         Triangulation<dim, spacedim2> &      out_tria);
 
+  /**
+   * Convert a triangulation consisting only of hypercube cells
+   * (quadrilaterals, hexahedrons) to a triangulation only consisting of
+   * simplices (triangles, tetrahedra).
+   * Thereby, in 2D a quadrilateral cell is converted into 8 triangle cells,
+   * whereas in 3D a hexahedron cell is converted into 24 tetrahedra cells.
+   *
+   * Material ID and boundary IDs will be inherited after conversion.
+   *
```

Comment:
For the documentation, it would be nice to have the picture you have at the top of the PR in here as well. To do this, add the .png to `doc/doxygen/images` and reference it like this: https://github.com/dealii/dealii/blob/master/include/deal.II/lac/solver.h#L325
which will then show up like this (scroll down to the pictures):
https://dealii.org/developer/doxygen/deal.II/classSolverBase.html